Piece.cpp

```diff
@@ -12,43 +12,89 @@
 #include <cmath>
 
 
-Piece::Piece(Basepiece basepiece, sf::Vector2i sectorPosition) {
-  this->basepiece = basepiece;
-  this->sectorPosition = sectorPosition;
+Piece::Piece(Game &game,  Board &board, Basepiece basepiece, sf::Vector2i sectorPosition, int player) :
+  player(player),
+  basepiece(basepiece) {
+  facing_front = !player;
+  consumeBasepiece(basepiece);
+  switch (basepiece){
+    case Basepiece::KING:
+      moveset.offsets.push_back(sf::Vector2i( 1, 0));
+      moveset.offsets.push_back(sf::Vector2i( 0, 1));
+      moveset.offsets.push_back(sf::Vector2i(-1, 0));
+      moveset.offsets.push_back(sf::Vector2i( 0,-1));
+      break;
+    case Basepiece::QUEEN:
+      moveset.diagonal = true;
+      moveset.horizontal = true;
+      moveset.vertical = true;
+      break;
+    case Basepiece::PAWN:
+      moveset.offsets.push_back(sf::Vector2i( 0, player ? -1 : 1));
```

Comment:
thinking about it, pawns can move diagonally to kill an enemy piece. I'm wondering how this will be handled @sweetsbeats @Nopey

Reply:
There is a lot of missing functionality from our chess, including castling, promotion, the weird pawn double move thing i never understood..
My recommendation is to not worry about it